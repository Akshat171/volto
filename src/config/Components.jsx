import { PreviewImage } from '@plone/volto/components';
import TeaserDefaultBody from '@plone/volto/components/manage/Blocks/Teaser/DefaultBody';

export const components = {
  PreviewImage: { component: PreviewImage },
  'Teaser|Default': { component: TeaserDefaultBody },
};
