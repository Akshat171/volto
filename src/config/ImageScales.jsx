export const imageScales = {
  huge: 1600,
  great: 1200,
  larger: 1000,
  large: 800,
  teaser: 600,
  preview: 400,
  mini: 200,
  thumb: 128,
  tile: 64,
  icon: 32,
};
