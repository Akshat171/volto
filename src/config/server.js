import imagesMiddleware from '@plone/volto/express-middleware/images';
import filesMiddleware from '@plone/volto/express-middleware/files';
import robotstxtMiddleware from '@plone/volto/express-middleware/robotstxt';
import sitemapMiddleware from '@plone/volto/express-middleware/sitemap';
import devProxyMiddleware from '@plone/volto/express-middleware/devproxy';
import prefixPathMiddleware from '@plone/volto/express-middleware/prefixPath';

const settings = {
  expressMiddleware: [
    devProxyMiddleware(),
    prefixPathMiddleware(),
    filesMiddleware(),
    imagesMiddleware(),
    robotstxtMiddleware(),
    sitemapMiddleware(),
  ],
  criticalCssPath: 'public/critical.css',
  readCriticalCss: null, // so it will be defaultReadCriticalCss
  extractScripts: { errorPages: false },
};

export default settings;
