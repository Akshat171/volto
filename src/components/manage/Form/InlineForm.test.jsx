import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import config from '@plone/volto/registry';

import InlineForm from './InlineForm';

const mockStore = configureStore();

function NewBaseWidget(name) {
  return (props) => (
    <div id={`mocked-field-${props.id}`} className={`mocked-${name}-widget`}>
      {props.title || 'No title'} - {props.description || 'No description'}
      {props.value}
    </div>
  );
}

beforeAll(() => {
  config.widgets.default = NewBaseWidget('default');
});

describe('Form', () => {
  it('renders a form component', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });

    let formData = {};

    const { container } = render(
      <Provider store={store}>
        <InlineForm
          schema={{
            fieldsets: [
              {
                id: 'default',
                title: 'Default',
                fields: ['title'],
              },
            ],
            properties: {
              title: {
                title: 'The title',
              },
              title2: {
                title: 'The title 2',
              },
            },
            required: [],
          }}
          onSubmit={() => {}}
          onCancel={() => {}}
          formData={formData}
          onChangeBlock={(block, data) => {
            formData = data;
          }}
        />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
  it('renders a form component with defaults in the schema', () => {
    const store = mockStore({
      intl: {
        locale: 'en',
        messages: {},
      },
    });
    let formData = {};

    const { container } = render(
      <Provider store={store}>
        <InlineForm
          schema={{
            fieldsets: [
              {
                id: 'default',
                title: 'Default',
                fields: ['title'],
              },
            ],
            properties: {
              title: {
                title: 'The title',
                default: 'This is the default',
              },
            },
            required: [],
          }}
          onSubmit={() => {}}
          onCancel={() => {}}
          onChangeBlock={(block, data) => {
            formData = data;
          }}
          formData={formData}
        />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
