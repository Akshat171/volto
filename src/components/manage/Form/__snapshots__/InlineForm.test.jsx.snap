// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders a form component 1`] = `
<div>
  <div
    class="ui form"
  >
    <div
      id="blockform-fieldset-default"
    >
      <div
        class="ui segment form attached"
      >
        <div
          class="mocked-default-widget"
          id="mocked-field-title"
        >
          The title
           - 
          No description
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Form renders a form component with defaults in the schema 1`] = `
<div>
  <div
    class="ui form"
  >
    <div
      id="blockform-fieldset-default"
    >
      <div
        class="ui segment form attached"
      >
        <div
          class="mocked-default-widget"
          id="mocked-field-title"
        >
          The title
           - 
          No description
          This is the default
        </div>
      </div>
    </div>
  </div>
</div>
`;
